portfolio/src/main/webapp/script.js

```diff
@@ -40,20 +40,35 @@ function addRandomQuote() {
 
 /** Adds messages (data) to website */
 function addData(){
-    fetch("/data")
+    var loggedIn = false;
+    fetch("/login")
         .then(response => response.json())
-        .then((cmts) =>{
-            const dataContainer = document.getElementById("data-container");
-            dataContainer.innerHTML = '';
-            cmts.forEach( cmt => {
-                dataContainer.appendChild(createListElement(cmt));  
-            });
+        .then( (authentication) => {
+            loggedIn = authentication.status;
+            if (loggedIn){
+                fetch("/data")
+                    .then(response => response.json())
+                    .then((comments) =>{
+                        const dataContainer = document.getElementById("data-container");
+                        dataContainer.innerHTML = '';
+                        comments.forEach( comment => {
+                            dataContainer.appendChild(createListElement(comment));  
+                        });
+                    });
+                document.getElementById("login").innerHTML = "<p>Logout <a href=\"" + authentication.logout + "\">here</a>.</p>";
+            }
+            else{
+                document.getElementById("form").style.display="none";
```

Comment:
Hiding a form based on style only is fine here but anyone can re-display it in their browser with something like Chrome Dev Tools. Something to be aware of.